confidence-core/src/main/java/org/saynotobugs/confidence/quality/function/MapsArgument.java

```diff
@@ -0,0 +1,77 @@
+package org.saynotobugs.confidence.quality.function;
+
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+import org.dmfs.srcless.annotations.staticfactory.StaticFactories;
+import org.saynotobugs.confidence.Assessment;
+import org.saynotobugs.confidence.Description;
+import org.saynotobugs.confidence.Quality;
+import org.saynotobugs.confidence.assessment.FailUpdated;
+import org.saynotobugs.confidence.description.Spaced;
+import org.saynotobugs.confidence.description.Text;
+import org.saynotobugs.confidence.description.Value;
+
+/**
+ * This quality can be used to isolate the logic of the outer function in a delegating function,
+ * which enables testing only the actual functions logic, not the logic of the delegate function.
+ *
+ * <pre>
+ * OuterFunction
+ * +-----------------------------------------------------------------------------------+
+ * |                      DelegateFunction                                             |
+ * |                     +------------------------------------------+                  |
+ * | OuterArgument ----&gt; |  DelegateArgument -----&gt; DelegateResult  | ---&gt; OuterResult |
+ * |                     +------------------------------------------+                  |
+ * +-----------------------------------------------------------------------------------+
+ * </pre>
+ */
+@StaticFactories(value = "Core", packageName = "org.saynotobugs.confidence.quality")
+public final class MapsArgument<In, DelegateIn, Out> implements Quality<Function<Function<DelegateIn, Out>, Function<In, Out>>> {
+	private final In mOuterArgument;
+	private final Out mDelegateResult;
+	private final Quality<? super DelegateIn> mDelegateArgumentQuality;
+
+	/**
+	 * Creates new instance.
+	 *
+	 * @param outerArgument  outer argument
+	 * @param delegateResult delegate result, no impact on quality, but must be a valid result that does not cause an excaption
+	 * @param delegateArgumentQuality quality verifying the delegate argument
+	 */
+	public MapsArgument(In outerArgument, Out delegateResult, Quality<DelegateIn> delegateArgumentQuality) {
+		this.mOuterArgument = outerArgument;
+		// we need a concrete valid result of the delegate, else when the outer function wants to manipulate
+		// the result, and we just return "null", there will be a NPE
+		this.mDelegateResult = delegateResult;
+		this.mDelegateArgumentQuality = delegateArgumentQuality;
+	}
+
+	@Override
+	public Assessment assessmentOf(Function<Function<DelegateIn, Out>, Function<In, Out>> candidate) {
+		final AtomicReference<DelegateIn> delegateValueReference = new AtomicReference<>();
+		final Function<In, Out> actualFunction = candidate.apply(delegateValue -> {
+			delegateValueReference.set(delegateValue);
+			return mDelegateResult;
```

Comment:
let's take the shortcut here and throw a custom `Error`. Having to provide a return value makes this more cumbersome.